src/main/java/org/assertj/core/api/AbstractAssert.java

```diff
@@ -358,105 +360,112 @@ private void printDescriptionText() {
   /** {@inheritDoc} */
   @Override
   public SELF isEqualTo(Object expected) {
-    objects.assertEqual(info, actual, expected);
+    //return doAssert(innerExpected -> objects.assertEqual(info, actual, innerExpected), expected);
+    return doAssert(() -> objects.assertEqual(info, actual, expected));
+  }
+
+  // Using this method means that we have to create a new object always, in order to use the input parameter
+  protected SELF doAssert(Runnable assertRunnable) {
+    if (ignoreWhen.test(actual)) {
+      return myself;
+    }
+
+    assertRunnable.run();
+
+    return myself;
+  }
+
+  // Using this method we can avoid not needed creation of objects, but passing everything we need to the consumer
```

Comment:
which object are we talking about?
we still need to build a consumer so it does not feel different that creating a Runnable.

Reply:
I think that the compiler can infer whether we need to create a new consumer every time or not. If we don't use parameters outside of the consumer, I think that the compiler can infer that. I might be wrong though